File: tests/checking/cases/port_receiver.cpp
// The same mistake, with the requirement WRITTEN as a concept.
#include <concepts>
template <class T> concept Readable = requires(T t) { t.read(); };
struct File { void read() {} };
template <Readable T> void inner(T t) { t.read(); }
template <Readable T> void outer(T t) { inner(t); }
int main() {
    long n = 0;
    outer(n);                                   // MISTAKE
}

File: tests/checking/cases/port_receiver.zig
// The same mistake. `anytype` is unconstrained, which is the point of the row.
const File = struct {
    fn read(_: File) void {}
};
fn inner(t: anytype) void { t.read(); }
fn outer(t: anytype) void { inner(t); }
pub fn main() void {
    const n: i64 = 0;
    outer(n);                                   // MISTAKE
}
